### PR TITLE
Item distribution is now done via the public post

### PR DESCRIPTION
**Pull request by annando**
This is a preparation for the rework of the item handling. We will now store every public post with the "uid=0" at first and then create the posts for the different users.

We started with Diaspora posts. Additionally the endpoint for public DFRN posts is now working. Some time later we will implement the sending of DFRN posts via the relay.

### CODE REVIEW

File: src/Model/Item.php [hunk 1]
```diff
@@ -839,6 +839,81 @@ public static function insert($item, $force_parent = false, $notify = false, $do
...
+	}
+
+	/**
+	 * @brief Store public items for the receivers
+	 *
+	 * @param integer $itemid  Item ID that should be added
+	 * @param array   $item    The item entry that will be stored
+	 * @param array   $contact The contact that will receive the item entry
+	 */
+	private static function storeForUser($itemid, $item, $contact)
```
Comment (MrPetovan) [inline src/Model/Item.php]: This method doesn’t references the `user` table, shouldn’t it be named `storeForConact` instead?

File: src/Protocol/DFRN.php [hunk 2]
```diff
@@ -2773,6 +2773,10 @@ private static function processEntry($header, $xpath, $entry, $importer, $xml)
 			if ($posted_id) {
 				logger("Reply from contact ".$item["contact-id"]." was stored with id ".$posted_id, LOGGER_DEBUG);
 
+				IF ($item['uid'] == 0) {
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php]: Wrong capitalization for the `if`.

File: src/Model/Item.php (hunk 1 above)
Comment (annando) [reply src/Model/Item.php]: It uses the contact table to find the user.

File: src/Protocol/DFRN.php (hunk 2 above)
Comment (annando) [reply src/Protocol/DFRN.php]: Ups ...

File: src/Model/Item.php (hunk 1 above)
Comment (annando) [reply src/Model/Item.php]: I clarified the whole stuff. The contact array wasn't needed - only the uid.

**Pull request merged**